packages/libp2p/src/connection-manager/auto-dial.ts

```diff
@@ -207,9 +212,26 @@ export class AutoDial implements Startable {
       return 0
     })
 
-    log('selected %d/%d peers to dial', sortedPeers.length, peers.length)
+    const peersThatHaveNotFailed = sortedPeers.filter(peer => {
+      const lastDialFailure = peer.metadata.get(LAST_DIAL_FAILURE_KEY)
 
-    for (const peer of sortedPeers) {
+      if (lastDialFailure == null) {
+        return true
+      }
+
+      const when = parseInt(uint8ArrayToString(lastDialFailure))
+
+      if (isNaN(when)) {
+        return true
+      }
+
+      // only dial if the time since the last failure is above the retry threshold
+      return Date.now() - when > this.autoDialPeerRetryThresholdMs
```

Comment:
```suggestion
      const lastDialFailuretimestamp = parseInt(uint8ArrayToString(lastDialFailure))

      if (isNaN(lastDialFailuretimestamp)) {
        return true
      }

      // only dial if the time since the last failure is above the retry threshold
      return Date.now() - lastDialFailuretimestamp > this.autoDialPeerRetryThresholdMs
```

Reply:
a more intentional variable name here would be nice